File: 07-architecture-for-team/android/app/src/main/java/com/chucknorrisviewer/AppComponent.java
package com.chucknorrisviewer;

import com.chucknorrisviewer.react.MyReactActivity;
import com.chucknorrisviewer.react.ReactModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = { AppModule.class, ReactModule.class })
public interface AppComponent {
    void inject(MainApplication app);
    void inject(MyReactActivity myReactActivity);
}

File: 06-flux-architecture/flux-rxjava/app/src/main/java/jp/satorufujiwara/flux/di/AppComponent.kt
package jp.satorufujiwara.flux.di

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import jp.satorufujiwara.flux.App
import javax.inject.Singleton

@Singleton
@Component(modules = [
  AndroidSupportInjectionModule::class,
  AppModule::class,
  UiModule::class]
)
interface AppComponent : AndroidInjector<App> {

  @Component.Builder
  interface Builder {
    @BindsInstance
    fun application(application: App): Builder

    fun build(): AppComponent
  }

  override fun inject(app: App)
}